crud/request.go

```diff
@@ -0,0 +1,109 @@
+package crud
+
+import (
+	"context"
+
+	"github.com/tarantool/go-tarantool"
+)
+
+type baseCrudRequest struct {
+	impl *tarantool.CallRequest
+}
+
+// Context sets a passed context to CRUD request.
+func (req *baseCrudRequest) Context(ctx context.Context) *baseCrudRequest {
+	req.impl = req.impl.Context(ctx)
+
+	return req
+}
+
+// Code returns IPROTO code for CRUD request.
+func (req *baseCrudRequest) Code() int32 {
+	return req.impl.Code()
+}
+
+// Ctx returns a context of CRUD request.
+func (req *baseCrudRequest) Ctx() context.Context {
+	return req.impl.Ctx()
+}
+
+// Async returns is CRUD request expects a response.
+func (req *baseCrudRequest) Async() bool {
+	return req.impl.Async()
+}
+
+type spaceCrudRequest struct {
+	baseCrudRequest
+	space interface{}
+}
+
+func (req *spaceCrudRequest) setSpace(space interface{}) {
+	req.space = space
+}
+
+// CrudInsertRequest helps you to create an CRUD insert
+// request object for execution by a Connection.
+type CrudInsertRequest struct {
+	spaceCrudRequest
+	tuple interface{}
+}
+
+// NewCrudInsertRequest returns a new empty CrudInsertRequest.
+func NewCrudInsertRequest(space interface{}) *CrudInsertRequest {
```

Comment:
I think that's enough:

```suggestion
func NewInsertRequest(space interface{}) *InsertRequest {
```

In user code, there will still be a difference from a package prefix:
```Go
req := tarantool.NewInsertRequest(foo)
сrudReq := crud.NewInsertRequest(bar)
```

Reply:
Fixed.